rust/gui-client/docs/launch-modes.md

```diff
@@ -0,0 +1,50 @@
+# Linux launch modes
+
+(Without thinking of systemd yet)
+
+1. No GUI allowed
+2. GUI spawns its own tunnel subprocess
+3. GUI and tunnel start up separately, GUI connects to the tunnel
```

Comment:
Sure maybe we should focus on (3) first. (2) requires sudo so it's like, if you're a power user with sudo powers, you could just use the CLI, or you would have the expertise to configure Mode 3 anyway.

The only reason I was thinking of starting with Mode 2 is that it might be simpler to test, and the security is simpler (The processes trust each other based on a parent-child relationship, same as web browsers do) But customers probably won't want Mode 2, they'll probably need Mode 3.